pkg/pubsub/amqp/amqppubsub.go

```diff
@@ -240,3 +241,75 @@ func newBackOff() backoff.BackOff {
 
 	return b
 }
+
+type subscriberInfo struct {
+	subscriber    subscriber
+	subscriptions int
+}
+
+type subscriberConnectionMgr struct {
+	createSubscriber  subscriberFactory
+	mutex             sync.RWMutex
+	subscribers       []*subscriberInfo
+	current           *subscriberInfo
+	subscriptionLimit int
+}
+
+func newSubscriberMgr(limit int, factory subscriberFactory) *subscriberConnectionMgr {
+	return &subscriberConnectionMgr{
+		subscriptionLimit: limit,
+		createSubscriber:  factory,
+	}
+}
+
+func (m *subscriberConnectionMgr) Close() error {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+
+	logger.Infof("Closing %d subscriber connections", len(m.subscribers))
+
+	for _, s := range m.subscribers {
+		if err := s.subscriber.Close(); err != nil {
+			logger.Warnf("Error closing subscriber: %s", err)
```

Comment:
Shouldn't this be `logger.Errorf`?

Reply:
I made it a warning so as not to alarm the Operations team. We're shutting down anyway.